### PR TITLE
docs: document documentation process

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

File: docs/practices/docs.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# Documentation
+
+This chapter describes nearcore approach to develope documentation. In general,
```
Comment (akhi3030) [inline docs/practices/docs.md:3]:
```suggestion
This chapter describes nearcore's approach to documentation. In general,
```

File: docs/practices/docs.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+there are three primary types of documentation to keep in mind:
```
Comment (akhi3030) [inline docs/practices/docs.md:4]:
```suggestion
there are three primary types of documentations to keep in mind:
```

File: docs/practices/docs.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+* **Protocol specification** ([source](https://github.com/near/NEPs),
+  [rendered](https://nomicon.io)) is a formal description of the NEAR protocol.
+  It can be used to implement alternative NEAR clients.
+* **User Docs** ([rendered](https://docs.near.org)) explain how to use the near
+  network. User docs are also split into documentation for validators (that is,
+  how to run your own near node) and documentation for smart contract
+  developers.
+* **Internal Development Docs** ([rendered](https://near.github.io/nearcore/),
+  [source](https://github.com/near/nearcore/tree/master/docs)) is the book you
+  are reading right now! The target audience here are developers of nearcore.
```
Comment (akhi3030) [inline docs/practices/docs.md:15]: nit: I think it is easy to misread "developers of nearcore" and deduce something like developers building on top of near.  I saw this happen in a couple of conversations.  I would rephrase it like something below

```suggestion
  are reading right now! The target audience here are nearcore contributors.
```

File: docs/practices/docs.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
+# Documentation
+
+This chapter describes nearcore approach to develope documentation. In general,
+there are three primary types of documentation to keep in mind:
+
+* **Protocol specification** ([source](https://github.com/near/NEPs),
+  [rendered](https://nomicon.io)) is a formal description of the NEAR protocol.
+  It can be used to implement alternative NEAR clients.
+* **User Docs** ([rendered](https://docs.near.org)) explain how to use the near
```
Comment (akhi3030) [inline docs/practices/docs.md:9]:
```suggestion
* **User docs** ([rendered](https://docs.near.org)) explain how to use the near
```

File: docs/practices/docs.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  network. User docs are also split into documentation for validators (that is,
+  how to run your own near node) and documentation for smart contract
+  developers.
+* **Internal Development Docs** ([rendered](https://near.github.io/nearcore/),
```
Comment (akhi3030) [inline docs/practices/docs.md:13]:
```suggestion
* **Internal development docs** ([rendered](https://near.github.io/nearcore/),
```

File: docs/practices/docs.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  [source](https://github.com/near/nearcore/tree/master/docs)) is the book you
+  are reading right now! The target audience here are developers of nearcore.
+
+Here, we focus on the last sort -- internal docs.
```
Comment (akhi3030) [inline docs/practices/docs.md:17]: I think we can remove this line.  We say above that this book is about internal docs.

```suggestion
```

File: docs/practices/docs.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+The book deployed via this GitHub Action:
+[.github/workflows/book.yml](https://github.com/near/nearcore/blob/master/.github/workflows/book.yml).
+It just runs mdBook and then deploys the result to
+https://near.github.io/nearcore/.
+
+For internal docs, you often want to have pretty pictures. We don't currently
+have a recommended workflow, but here are some tips:
+
+* Don't add binary media files to Git. Rather, upload images as comments to this
```
Comment (akhi3030) [inline docs/practices/docs.md:85]: nit: might be worth it to add a sentence to explain why we do not linke binary files.

**Review by akhi3030 — APPROVED** [review]

File: docs/practices/docs.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+  super-secret issue [#7821](https://github.com/near/nearcore/issues/7821), and
+  then link to the images as
```
Comment (matklad) [inline docs/practices/docs.md:87]:
```suggestion
* Don't add binary media files to Git to avoid inflating repository size. 
  Rather, upload images as comments to this super-secret issue 
  [#7821](https://github.com/near/nearcore/issues/7821), and then link to 
  the images as
```
